apigw/src/shared/routes/saml.ts

```diff
@@ -206,20 +272,33 @@ export default function createSamlRouter(
   //   request.
   router.get(
     `/logout/callback`,
-    logoutCallback,
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-    passport.authenticate(strategyName),
-    (req, res) =>
-      res.redirect(parseRelayState(req) ?? endpointConfig.defaultPageUrl)
+    logoutCallback(async (req) => {
+      const originalQuery = url.parse(req.url).query ?? ''
+      const { profile, loggedOut } = await saml.validateRedirectAsync(
+        req.query,
+        originalQuery
+      )
+      if (!loggedOut) {
+        throw new Error('Invalid SAML message type: expected logout response')
+      }
+      return profile
+    })
   )
   router.post(
     `/logout/callback`,
     urlencodedParser,
-    logoutCallback,
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-    passport.authenticate(strategyName),
-    (req, res) =>
-      res.redirect(parseRelayState(req) ?? endpointConfig.defaultPageUrl)
+    logoutCallback(async (req) => {
+      const { profile, loggedOut } =
+        'SAMLRequest' in req.body
```

Comment:
Tähän voisi lisätä hieman dokumentaatiota ympärille, että mitä tapauksia tämä käsittelee. Jos ymmärsin oikein niin esim. näin:

```suggestion
// this handler can be invoked at the end of a IDP-initiated logout flow or SP-initiated (our app) logout flow
        isIdpInitiatedLogout(req.body)
```
ja 
```typescript
? // IDP-initiated logout - IDP is requesting we logout
await saml.validatePostRequestAsync(req.body)
: // SP-initiated logout - IDP is responding to our logout request
await saml.validatePostResponseAsync(req.body)
```

Vai olisiko nämä mahdollista erottaa kahdeksi eri routeksi?

```typescript
router.post(
    `/logout/idp`,
    ...
)

router.post(
    `/logout/callback`,
    ...
)
```

Reply:
Hyvä idea mutta tämä vaatisi metadatojen uudelleenlähetyksen jokaiseen saml-integraatioon jokaisessa kunnassa. Oikeasti paras design koko hommalle omasta mielestä olisi:

- `/login` (nykyinen)
- `/logout` (nykyinen)
- `/callback` (hoitaa sekä login- että logout responset)
- `/idp-logout` (hoitaa idp:n logout requestin)